### PR TITLE
Output object instead of dataframe to handle sorting and filtering

### PR DESCRIPTION
**Pull request by rajasbansal**
1. Add an object as the result of run instead of a dataframe
2. The object contains the dataframe of labels as a field
3. Filter and eval according to the object

### CODE REVIEW

File: src/autolabel/labeling_output.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        if filter_func:
+            filtered_df = filtered_df.apply(filter_func, axis=1)
+
+        return LabelingOutput(
+            self.config,
+            filtered_df,
+            self.metrics,
+        )
+
+    def errors(self):
```
Comment (nihit) [inline src/autolabel/labeling_output.py:114]: errors vs mistakes is a bit confusing naming - how about we roll it into the same function? 

llm label != gt label is ultimately a kind of error?

File: src/autolabel/labeling_output.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        filtered_df = self.df[self.df[self.prefix + "error"].notnull()]
+        return LabelingOutput(
+            self.config,
+            filtered_df,
+            self.metrics,
+        )
+
+    def non_errors(self):
```
Comment (nihit) [inline src/autolabel/labeling_output.py:122]: can remove this for now

File: src/autolabel/labeling_output.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+        filtered_df = self.df[
+            self.df[self.prefix + "label"] == self.df[gt_label_column]
+        ]
+        return LabelingOutput(
+            self.config,
+            filtered_df,
+            self.metrics,
+        )
+
+    def correct_and_confident(self, threshold: float = 0.5):
```
Comment (nihit) [inline src/autolabel/labeling_output.py:169]: Suggest simplifying this to only filter by confidence (independent of correct)

**Review by nihit — APPROVED** [review]
lgtm otherwise!

File: src/autolabel/labeling_output.py (hunk 1 above)
Comment (rajasbansal) [reply src/autolabel/labeling_output.py:114]: I wanted to separate them out in order to find the examples where the model was not able to answer, if potentially they went over context length or some other input specific problem and we wanted to fix them

**Pull request merged**